packages/reports/addon/components/common-actions/schedule.js

```diff
@@ -118,8 +126,11 @@ export default Ember.Component.extend({
         });
 
         set(this, 'attemptedSave', false);
-        this.attrs
-          .onSave(deliveryRule)
+        let savePromise = new RSVP.Promise((resolve, reject) => {
```

Comment:
Why do we need the promise wrapper?

Reply:
The save is being handled through a consumer now. This promise will resolve or reject once the save succeeds or fails, then we can push the appropriate notification from this component.